File: content-solutions/hash/snack-pair.js
function solution(prices, budget) {
  const seen = new Map();
  for (let j = 0; j < prices.length; j++) {
    const need = budget - prices[j];
    if (seen.has(need)) return [seen.get(need), j];
    if (!seen.has(prices[j])) seen.set(prices[j], j);
  }
  return [];
}

File: content-solutions/hash/snack-pair.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices, int budget) {
        Map<Integer, Integer> seen = new HashMap<>();
        for (int j = 0; j < prices.length; j++) {
            int need = budget - prices[j];
            if (seen.containsKey(need)) return new int[] {seen.get(need), j};
            seen.putIfAbsent(prices[j], j);
        }
        return new int[0];
    }
}
